src/schemas/json/bacalhau-job.json

```
@@ -0,0 +1,506 @@
+{
+  "$defs": {
+    "Deal": {
+      "properties": {
+        "Concurrency": {
+          "type": "integer"
+        },
+        "Confidence": {
+          "type": "integer"
+        },
+        "MinBids": {
+          "type": "integer"
+        }
+      },
+      "additionalProperties": false,
+      "type": "object"
+    },
+    "Job": {
+      "properties": {
+        "APIVersion": {
+          "type": "string"
+        },
+        "Metadata": {
+          "$ref": "#/$defs/Metadata"
+        },
+        "Spec": {
+          "$ref": "#/$defs/Spec"
+        },
+        "Status": {
+          "$ref": "#/$defs/JobStatus"
+        }
+      },
+      "additionalProperties": false,
+      "type": "object",
+      "required": ["APIVersion"]
+    },
+    "JobEvent": {
+      "properties": {
+        "APIVersion": {
+          "type": "string"
+        },
+        "JobID": {
+          "type": "string"
+        },
+        "ShardIndex": {
+          "type": "integer"
+        },
+        "ClientID": {
+          "type": "string"
+        },
+        "SourceNodeID": {
+          "type": "string"
+        },
+        "TargetNodeID": {
+          "type": "string"
+        },
+        "EventName": {
+          "type": "integer"
+        },
+        "Spec": {
+          "$ref": "#/$defs/Spec"
+        },
+        "JobExecutionPlan": {
+          "$ref": "#/$defs/JobExecutionPlan"
+        },
+        "Deal": {
+          "$ref": "#/$defs/Deal"
+        },
+        "Status": {
+          "type": "string"
+        },
+        "VerificationProposal": {
+          "type": "string",
+          "contentEncoding": "base64"
+        },
+        "VerificationResult": {
+          "$ref": "#/$defs/VerificationResult"
+        },
+        "PublishedResult": {
+          "$ref": "#/$defs/StorageSpec"
+        },
+        "EventTime": {
+          "type": "string",
+          "format": "date-time"
+        },
+        "SenderPublicKey": {
+          "$ref": "#/$defs/PublicKey"
+        },
+        "RunOutput": {
+          "$ref": "#/$defs/RunCommandResult"
+        }
+      },
+      "additionalProperties": false,
+      "type": "object"
+    },
+    "JobExecutionPlan": {
+      "properties": {
+        "ShardsTotal": {
+          "type": "integer"
```

```suggestion
          "type": "integer",
          "minimum": 0
```

Optional fix: `minimum` can be added to restrict range of valid values.